src/relay/qnn/util.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file src/relay/qnn/util.h
+ * \brief Utility methods needs for quantized ops that can be shared
+ */
+
+#ifndef TVM_RELAY_QNN_UTIL_H_
+#define TVM_RELAY_QNN_UTIL_H_
+
+#include <tvm/expr.h>
+#include <tvm/relay/expr.h>
+#include <limits>
+
+namespace tvm {
+namespace relay {
+namespace qnn {
+
+static inline bool IsQNNDataType(const DataType& dtype) {
+  return dtype == Int(8) || dtype == UInt(8)
+      || dtype == Int(16) || dtype == UInt(16)
+      || dtype == Int(32);
```

Comment:
I suggest that we re-evaluate whether do we need 32 bit quantization support. Let's discuss in the RFC thread to loop more people.

Reply:
Is there any particular reason we should not include Int(32) for requantize?

I think for requantize, the in_dtype can be `int32`. For out_dtype, it seems unlikely that it will be `int32`.

Reply:
I think requantize is fine as the intermediate results are int32. My consideration is quantize and requantize. I will leave comments to the RFC #3591 later :)

Reply:
see https://github.com/dmlc/tvm/issues/3591#issuecomment-515008567